pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -715,8 +715,14 @@ impl ConnectionManagerState {
         &self,
         node_to_omit: Option<NodeId>,
     ) -> Option<(NodeId, &SafekeeperTimelineInfo, PgConnectionConfig)> {
+        let currently_at = self.timeline.get_last_record_lsn();
         self.applicable_connection_candidates()
             .filter(|&(sk_id, _, _)| Some(sk_id) != node_to_omit)
+            .filter(|&(sk_id, info, _)| {
```

Comment:
you don't use sk_id here

Reply:
Ah, true :facepalm:

Reply:
[b7c7a4e](https://github.com/neondatabase/neon/pull/4366/commits/b7c7a4ef892715cd45be44b9e3bb5defbd609483)